core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -81,10 +81,10 @@ def start_navigator(self) -> None:
         # self.start_mavlink_manager(Endpoint(local_endpoint))
 
     def start_serial(self, device: str) -> None:
-        self.start_mavlink_manager(Endpoint("serial", device, 115200))
+        self.start_mavlink_manager(Endpoint("Master", self.settings.app_name, False, True, "serial", device, 115200))
```

Comment:
This is really bad as API, can we have a scruct or anything that provides explicit names for the fields ?
"something", something, True, False is not what is each field.

Reply:
I would recommend to read:
https://wiki.qt.io/API_Design_Principles
https://doc.qt.io/archives/qq/qq13-apis.html
https://harrymoreno.com/assets/the-little-manual-of-api-design.pdf

Reply:
You mean to write it explicitly there?

```python
self.start_mavlink_manager(Endpoint(
    name="Master",
    owner=self.settings.app_name, 
    persistent=False, 
    protected=True,
    connection_type="serial",
    place=device,
    argument=115200
))
```

Reply:
Yes, but as you can see that makes the use really hard, I would recommend to break some of the extra configurations in a second function/struct to avoid unnecessary fields, and take some of the values as default on the constructor for the common use case.

Reply:
There are some common use cases, like the not-persistent/protected/APManager endpoint, the not-persistent/protected/"external-app" endpoint, etc. What we can do is to create builder methods for this. What do you think?

Reply:
> take some of the values as default on the constructor for the common use case.

This is a good suggestion that rafael has just implemented. Beyond that, I don't think that the code can be simplified or reduced on the caller side. These are necessary parameters to construct an endpoint. They cannot be configured in a second step like `set_owner`.

@patrickelectric If you see a better way, please suggest a patch on top of this.

Reply:
Will open an issue to discuss a new idea for this API.

Reply:
https://github.com/bluerobotics/companion-docker/issues/183